e2e_tests/tests/cluster/test_checkpoints.py

```diff
@@ -45,6 +48,75 @@ def test_gc_checkpoints_lfs() -> None:
     run_gc_checkpoints_test(exp.shared_fs_checkpoint_config())
 
 
+@pytest.mark.e2e_cpu
+def test_delete_checkpoints() -> None:
+    base_conf_path = conf.fixtures_path("no_op/single-default-ckpt.yaml")
+    config = conf.load_config(str(base_conf_path))
+    config["checkpoint_storage"] = {
+        "type": "shared_fs",
+        "host_path": "/tmp",
+        "storage_path": "delete-checkpoints-e2etest",
+    }
+    config["min_checkpoint_period"] = {"batches": 10}
+
+    exp_id_1 = exp.run_basic_test_with_temp_config(
+        config, model_def_path=conf.fixtures_path("no_op"), expected_trials=1
+    )
+
+    exp_id_2 = exp.run_basic_test_with_temp_config(
+        config, model_def_path=conf.fixtures_path("no_op"), expected_trials=1
+    )
+
+    test_session = exp.test_session()
+    exp_1_checkpoints = bindings.get_GetExperimentCheckpoints(
+        session=test_session, id=exp_id_1
+    ).checkpoints
+    exp_2_checkpoints = bindings.get_GetExperimentCheckpoints(
+        session=test_session, id=exp_id_1
+    ).checkpoints
+    assert len(exp_1_checkpoints) > 0, f"no checkpoints found in experiment with ID:{exp_id_1}"
+    assert len(exp_2_checkpoints) > 0, f"no checkpoints found in experiment with ID:{exp_id_2}"
+
+    indices_exp_1 = [i for i in range(0, len(exp_1_checkpoints))]
+    indices_exp_2 = [i for i in range(0, len(exp_2_checkpoints))]
+    d_exp_1_checkpoint_uuids = [
+        exp_1_checkpoints[d_index].uuid for d_index in random.sample(indices_exp_1, 2)
```

Comment:
I think instead you can just do `random.sample(range(len(exp_1_checkpoints)), 2)`